### PR TITLE
[Macros] Add OptionSet and plumb it through

### PR DESCRIPTION
**Pull request by DougGregor**

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

File: stdlib/public/core/Macros.swift [hunk 1]
```diff
@@ -60,3 +60,33 @@ public macro column<T: ExpressibleByIntegerLiteral>() -> T =
...
+///     @OptionSet<UInt8>
+///     struct ShippingOptions {
+///       private enum Options: Int {
+///         case nextDay
+///         case secondDay
+///         case priority
+///         case standard
+///       }
+///     }
+@attached(member)
```
Comment (hborla) [inline stdlib/public/core/Macros.swift:89]:
```suggestion
@attached(member, names: named(RawValue), named(rawValue), named(`init`), arbitrary)
```

**Comment by etcwilde** [comment]
@swift-ci please test

**Pull request merged**